engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -223,10 +224,20 @@ class DeleteWorkflowFilesActor(rootWorkflowId: RootWorkflowId,
     womValue.collectAsSeq({ case womSingleFile: WomSingleFile => womSingleFile })
   }
 
+  /**
+    * Returns Paths for WomSingleFiles in allOutputs that are not in finalOutputs, verifying that the Paths are contained
+    * in a workflow execution directory associated with the root workflow (as a sanity check to avoid deleting files
+    * outside of the workflow execution directory).
+    */
   def gatherIntermediateOutputFiles(allOutputs: Set[WomValue], finalOutputs: Set[WomValue]): Set[Path] = {
     val allOutputFiles = allOutputs.flatMap(getWomSingleFiles)
     val finalOutputFiles = finalOutputs.flatMap(getWomSingleFiles)
-    allOutputFiles.diff(finalOutputFiles).flatMap(toPath)
+    val potentialIntermediaries = allOutputFiles.diff(finalOutputFiles).flatMap(toPath)
+    val checkedIntermediaries = potentialIntermediaries.filter(p => rootWorkflowRoots.exists(r => p.pathAsString.startsWith(r.pathAsString)))
+    for ( path <- potentialIntermediaries.diff(checkedIntermediaries) ) {
+      log.warning(s"Did not delete $path because it is not contained within a workflow root directory for $rootWorkflowId")
```

Comment:
Should this be warning or info?

Reply:
Note that I verified the message is getting emitted when the centaur test runs:

````
no_input_delete complete. Final Outputs:
{
  "no_input_delete.done": true
}
2021-10-01 20:26:34,467 cromwell-system-akka.dispatchers.engine-dispatcher-12 INFO  - WorkflowExecutionActor-91b8fe05-ab57-4375-bb02-e5b5eceb05f4 [UUID(91b8fe05)]: Workflow no_input_delete_setup complete. Final Outputs:
{
  "no_input_delete_setup.done": "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/no_input_delete/test_execution/no_input_delete.txt: Creation time: Fri, 01 Oct 2021 20:17:34 GMT Update time: Fri, 01 Oct 2021 20:17:34 GMT Storage class: MULTI_REGIONAL Content-Length: 34 Content-Type: text/plain Hash (crc32c): utmokA== Hash (md5): 8cj2wBOOYQBaVlnfmlvYxw== ETag: CIb43oeEqvMCEAE= Generation: 1633119454608390 Metageneration: 1"
}
2021-10-01 20:26:34,493 cromwell-system-akka.actor.default-dispatcher-4 WARN  - Did not delete gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/no_input_delete/test_execution/no_input_delete.txt because it is not contained within a workflow root directory for 91b8fe05-ab57-4375-bb02-e5b5eceb05f4
2021-10-01 20:26:34,498 cromwell-system-akka.actor.default-dispatcher-4 INFO  - Root workflow 91b8fe05-ab57-4375-bb02-e5b5eceb05f4 does not have any intermediate output files to delete.
2021-10-01 20:26:34,504 cromwell-system-akka.dispatchers.engine-dispatcher-29 INFO  - WorkflowActor-91b8fe05-ab57-4375-bb02-e5b5eceb05f4 [UUID(91b8fe05)]: Successfully deleted intermediate output file(s) for root workflow 91b8fe05-ab57-4375-bb02-e5b5eceb05f4.

Reply:
I think it should be info. In the realm of the user's workflow, it is potentially a mistake - but in the realm of Cromwell the application that we monitor and maintain, it is working as designed.

Reply:
Done